ext/dom/php_dom.stub.php

```diff
@@ -2,1011 +2,1223 @@
+     */
+    const XML_ATTRIBUTE_ID = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_IDREF
+     */
+    const XML_ATTRIBUTE_IDREF = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_IDREFS
+     */
+    const XML_ATTRIBUTE_IDREFS = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_ENTITIES
+     */
+    const XML_ATTRIBUTE_ENTITY = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_NMTOKEN
+     */
+    const XML_ATTRIBUTE_NMTOKEN = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_NMTOKENS
+     */
+    const XML_ATTRIBUTE_NMTOKENS = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_ENUMERATION
+     */
+    const XML_ATTRIBUTE_ENUMERATION = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue XML_ATTRIBUTE_NOTATION
+     */
+    const XML_ATTRIBUTE_NOTATION = UNKNOWN;
 
-    /** @readonly */
-    public DOMNamedNodeMap $entities;
+    /**
+     * @var int
+     * @cvalue PHP_ERR
+     */
+    const DOM_PHP_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue INDEX_SIZE_ERR
+     */
+    const DOM_INDEX_SIZE_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue DOMSTRING_SIZE_ERR
+     */
+    const DOMSTRING_SIZE_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue HIERARCHY_REQUEST_ERR
+     */
+    const DOM_HIERARCHY_REQUEST_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue WRONG_DOCUMENT_ERR
+     */
+    const DOM_WRONG_DOCUMENT_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue INVALID_CHARACTER_ERR
+     */
+    const DOM_INVALID_CHARACTER_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue NO_DATA_ALLOWED_ERR
+     */
+    const DOM_NO_DATA_ALLOWED_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue NO_MODIFICATION_ALLOWED_ERR
+     */
+    const DOM_NO_MODIFICATION_ALLOWED_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue NOT_FOUND_ERR
+     */
+    const DOM_NOT_FOUND_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue NOT_SUPPORTED_ERR
+     */
+    const DOM_NOT_SUPPORTED_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue INUSE_ATTRIBUTE_ERR
+     */
+    const DOM_INUSE_ATTRIBUTE_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue INVALID_STATE_ERR
+     */
+    const DOM_INVALID_STATE_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue SYNTAX_ERR
+     */
+    const DOM_SYNTAX_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue INVALID_MODIFICATION_ERR
+     */
+    const DOM_INVALID_MODIFICATION_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue NAMESPACE_ERR
+     */
+    const DOM_NAMESPACE_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue INVALID_ACCESS_ERR
+     */
+    const DOM_INVALID_ACCESS_ERR = UNKNOWN;
+    /**
+     * @var int
+     * @cvalue VALIDATION_ERR
+     */
+    const DOM_VALIDATION_ERR = UNKNOWN;
 
-    /** @readonly */
-    public DOMNamedNodeMap $notations;
+    /** @alias DOM\DocumentType */
+    class DOMDocumentType extends DOMNode
```

Comment:
Is aliasing happening in the right direction? Should the namespaced variant be the canonical one?

Reply:
Making the namespaced variant canonical will silently break `get_class` & friends, I'd rather do that break after (if ever possible) we deprecate the old classes.

Reply:
Okay, then the references are a little inconsistent still:

- `DOM\Document::createAttribute()` returns `Attr` (i.e. `DOM\Attr`),
- but `DOM\XMLDocument::createEntityReference()` returns `DOMEntityReference` (which probably is `DOM\DOMEntityReference`. In fact the `@return` appears to be ignored, because of the `@implementation-alias`.